slice_test.go

```diff
@@ -20,3 +22,111 @@ func ExampleUniqueInterfaces() {
 	fmt.Println(u.UniqueInterfaces([]interface{}{13, "foo", "bar", 42, 13, 43, "baz"}))
 	// Output: [13 foo bar 42 43 baz]
 }
+
+func BenchmarkUniqueStrings(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data []string
+	}{
+		{"slice1", []string{"foofoo", "barbar", "foo", "bazbaz", "foo", "bar", "foobar", "baz", "foobaz"}},
+		{"slice2", bigSliceOfString()},
+	}
+
+	for _, bc := range cases {
+		b.Run(bc.Name, func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				u.UniqueStrings(bc.Data)
+			}
+		})
+		b.Run(bc.Name+"-parallel", func(b *testing.B) {
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					u.UniqueStrings(bc.Data)
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkUniqueInts(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data []int
+	}{
+		{"slice1", []int{4, 5, 2, 6, 7, 7, 2, 4, 2, 1, 1, 2, 3, 4}},
+		{"slice2", bigSliceOfInt()},
+	}
+
+	for _, bc := range cases {
+		b.Run(bc.Name, func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				u.UniqueInts(bc.Data)
+			}
+		})
+		b.Run(bc.Name+"-parallel", func(b *testing.B) {
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					u.UniqueInts(bc.Data)
+				}
+			})
+		})
+	}
+}
+
+func BenchmarkUniqueInterfaces(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data []interface{}
+	}{
+		{"slice1", []interface{}{1, 2, 2, 1, 3, 4, "foo", "bar", "foo", "foo", "barfoo"}},
+		{"slice2", bigSliceOfInterface()},
+	}
+
+	for _, bc := range cases {
+		b.Run(bc.Name, func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				u.UniqueInterfaces(bc.Data)
+			}
+		})
+		b.Run(bc.Name+"-parallel", func(b *testing.B) {
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					u.UniqueInterfaces(bc.Data)
+				}
+			})
+		})
+	}
+}
+
+func bigSliceOfInt() []int {
```

Comment:
these helpers should create duplicate entries sometimes since it's a "Unique" helper

Reply:
and it should be in a random-like order

ideally don't use random; you can maybe use something like this

https://play.golang.org/p/ebO0gC8ZowC

Reply:
ahah, my bad, it just generates a random-like order, but it actually generates no duplicates :)

Reply:
Use this one instead -> https://play.golang.org/p/jItPqkEpwzX

```console
 $ pbpaste| sort| uniq -c | sort -n  | awk '{print $1}' | uniq -c
    143 1
    124 2
     94 3
     45 4
     17 5
      2 6
      6 7
      1 8
```

Reply:
@moul Yes we will need to create them manually in for cycle :)

Reply:
you forgot to call `rand.Seed(42)` as in the example, which makes the benchmark inconsistent